### ISSUE TITLE
SeparableConv1D returns incorrect symbolic output shape for valid strided inputs

### ISSUE THREAD

**Issue by ningxiudg** [keras-team/keras#22496]
### Bug report

`keras.layers.SeparableConv1D` computes a correct output shape for eager tensors, but the symbolic output shape from `keras.Input()` is wrong when using `padding='valid'` and `strides > 1`.

### Reproduction
```python
import keras
import tensorflow as tf
import numpy as np

x_eager = tf.constant(np.random.rand(4, 10, 12), dtype=tf.float32)
x_sym = keras.Input(shape=(10, 12))

layer = keras.layers.SeparableConv1D(64, 11, strides=5, padding='valid', activation='relu')

y_eager = layer(x_eager)
y_sym = layer(x_sym)

print(y_eager.shape)
print(y_sym.shape)
```

### Expected result
Both outputs should report the same spatial shape.

### Actual result
- Eager output shape: `(4, 10, 64)`
- Symbolic output shape: `(None, 0, 64)`

This indicates incorrect static shape inference for `SeparableConv1D`.

**Comment by CarolinJoseph** [comment]
Hi, I’d like to work on this issue. Could you please assign it to me?

**Comment by ayash911** [comment]
Hi! Thanks for opening this issue. 

I was looking into fixing this, but when I ran your exact reproduction script, the eager and symbolic shapes matched perfectly. It seems like this bug might have already been resolved in a recent update. 

**My Environment:**
* **OS:** macOS (Apple Silicon)
* **Keras version:** 3.13.2
* **TensorFlow version:** 2.21.0

**My Output:**
(4, 0, 64)
(None, 0, 64)

Could you let me know which versions of Keras and TensorFlow you were using when you encountered this? If you upgrade to the latest version, does the issue still persist for you?
